icu4c/source/test/intltest/strcase.cpp

```diff
@@ -1314,7 +1314,8 @@ void StringCaseTest::TestCaseMapUTF8WithEdits() {
     Edits edits;
 
     int32_t length = CaseMap::utf8ToLower("tr", U_OMIT_UNCHANGED_TEXT,
-                                          reinterpret_cast<const char*>(u8"IstanBul"), 8, dest, UPRV_LENGTHOF(dest), &edits, errorCode);
+                                          reinterpret_cast<const char*>(u8"IstanBul"), 8,
```

Comment:
@tahonermann didn't comment on this line, but I assume this `reinterpret_cast` is no longer needed for the same reason as the one on line 1334. (But on first glance I don't see how line 1317 is testing anything different from line 1457. So maybe one of the pair is _supposed_ to be testing `const char*` and the other to be testing `const char8_t*`?)

Do you _want_ test coverage for both `char` and `char8_t`? If so, maybe consider making the whole test case `template<class CharT>`, reinterpret-casting all your string literals to `const CharT*`, and then instantiating the whole thing once for `char` and once for `char8_t`. I suspect that's overkill, personally, but I dunno.

Reply:
I don't think there's a char8_t overload for CaseMap::utf8ToLower(), so the cast remains necessary.

Reply:
As Tom found, we have two versions of the API, pointer+length vs. StringPiece. This unit test covers functionality and API.

https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1CaseMap.html

I don't intend to add char8_t and nullptr_t overloads everywhere. We might add some more, but probably not comprehensively.